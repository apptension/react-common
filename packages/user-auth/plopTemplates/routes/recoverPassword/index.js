export { default as RecoverPassword } from './recoverPassword.container';
