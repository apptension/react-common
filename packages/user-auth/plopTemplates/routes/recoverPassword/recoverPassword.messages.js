import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'recoverPassword.title',
    defaultMessage: 'Reset password',
  },
  emailLabel: {
    id: 'recoverPassword.fields.email.label',
    defaultMessage: 'Email',
  },
  emailPresenceError: {
    id: 'recoverPassword.fields.email.error.presence',
    defaultMessage: 'Email cannot be blank',
  },
  emailFormatError: {
    id: 'recoverPassword.fields.email.error.format',
    defaultMessage: 'Email is not valid',
  },
  emailMaxLengthError: {
    id: 'recoverPassword.fields.email.error.maxLength',
    defaultMessage: 'Email shouldn\'t be longer than 255 characters',
  },
  submit: {
    id: 'recoverPassword.submit',
    defaultMessage: 'Send',
  },
  sentMessage: {
    id: 'recoverPassword.sentMessage',
    defaultMessage: 'Check your email to get reset password link.',
  },
  resendEmail: {
    id: 'recoverPassword.resendEmail',
    defaultMessage: 'Resend email',
  },
});
