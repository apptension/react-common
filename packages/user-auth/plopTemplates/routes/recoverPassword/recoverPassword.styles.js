import styled from 'styled-components';

export const Container = styled.div``;

export const TextField = styled.input`
  flex: 1 1 auto;
`;

export const Form = styled.form`
  display: flex;
  flex-direction: column;
`;

export const ButtonGroup = styled.div`
  width: 100%;
`;

export const SuccessMessage = styled.div`
  flex: 1 1 auto;
  width: 100%;
  height: 100%;
  display: flex;
`;

export const SuccessMessageText = styled.div`
  width: 100%;
`;
