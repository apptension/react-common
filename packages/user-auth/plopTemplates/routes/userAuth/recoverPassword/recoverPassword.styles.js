import styled from 'styled-components';

export const SuccessMessage = styled.div`
  width: 100%;
`;

export const SuccessMessageText = styled.div`
  width: 100%;
`;
