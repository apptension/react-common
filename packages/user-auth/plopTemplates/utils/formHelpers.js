import reportError from 'report-error';
import { List } from "immutable";
import { when, is, pipe } from 'ramda';

export const getFormFieldHelperText = (intl, messages, { error, invalid, touched }) => {
  if (invalid && touched) {
    try {
      return intl.formatMessage(messages[pipe(
        when(
          is(List),
          (errors) => errors.first()
        ),
        when(
          is(Array),
          ([firstError]) => firstError
        ),
      )(error)]);
    } catch (e) {
      reportError(e);
    }
  }
  return '';
};
