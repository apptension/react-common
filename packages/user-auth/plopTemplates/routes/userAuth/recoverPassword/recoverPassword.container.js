import { reduxForm } from 'redux-form/immutable';
import validate from 'validate.js';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'ramda';
import { injectIntl } from 'react-intl';

import { RECOVER_PASSWORD_FORM, UserAuthActions, selectRecoverPasswordSuccess } from '../../../modules/userAuth';
import { RecoverPassword } from './recoverPassword.component';
import { selectLocalesLanguage } from '../../../modules/locales/locales.selectors';

const mapStateToProps = createStructuredSelector({
  recoverPasswordSuccess: selectRecoverPasswordSuccess,
  language: selectLocalesLanguage,
});

export const mapDispatchToProps = (dispatch) => bindActionCreators({
  recoverPassword: UserAuthActions.recoverPassword,
}, dispatch);

export default compose(
  injectIntl,
  connect(mapStateToProps, mapDispatchToProps),
  reduxForm({
    form: RECOVER_PASSWORD_FORM,
    validate: (values) => validate(values.toJS(), {
      email: {
        presence: {
          message: '^emailPresenceError',
        },
        email: {
          message: '^emailFormatError',
        },
        length: {
          maximum: 255,
          message: '^emailMaxLengthError',
        },
      },
    }),
  }),
)(RecoverPassword);
