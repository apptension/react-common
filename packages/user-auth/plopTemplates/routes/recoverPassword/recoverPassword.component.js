import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form/immutable';
import { FormattedMessage } from 'react-intl';
import { Link } from 'react-router-dom';

import { getFormFieldHelperText, renderWhenTrueOtherwise } from '../../utils/formHelpers';
import {
  Container,
  TextField,
  Form,
  ButtonGroup,
  SuccessMessage,
  SuccessMessageText,
} from './recoverPassword.styles';
import messages from './recoverPassword.messages';

export class RecoverPassword extends PureComponent {
  static propTypes = {
    intl: PropTypes.object.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    recoverPassword: PropTypes.func.isRequired,
    recoverPasswordSuccess: PropTypes.bool.isRequired,
  };

  handleSubmit = (values) => new Promise(() => {
    this.props.recoverPassword(
      values.get('email'),
    );
  });

  renderTextField = ({ meta, input, type, label }) => {
    return (
      <TextField
        error={meta.invalid && meta.touched}
        value={input.value}
        type={type}
        label={label}
        helperText={getFormFieldHelperText(this.props.intl, messages, meta)}
        InputProps={input}
      />
    );
  };

  renderForm = () => {
    const { intl, handleSubmit } = this.props;
    return (
      <Container>
        <Form onSubmit={handleSubmit(this.handleSubmit)} noValidate>
          <Field
            component={this.renderTextField}
            name="email"
            type="email"
            label={intl.formatMessage(messages.emailLabel)}
          />

          <ButtonGroup>
            <button type="submit">
              <FormattedMessage {...messages.submit} />
            </button>
          </ButtonGroup>
        </Form>
      </Container>
    );
  };

  renderSuccessMessage = () => {
    return (
      <SuccessMessage>
        <SuccessMessageText responsive>
          <FormattedMessage {...messages.sentMessage} />
        </SuccessMessageText>
        <Link to="/recover-password">
          <FormattedMessage {...messages.resendEmail} />
        </Link>
      </SuccessMessage>
    );
  };

  render() {
    const { recoverPasswordSuccess } = this.props;
    return (
      <Container>
        <FormattedMessage {...messages.title} />
        {renderWhenTrueOtherwise(this.renderSuccessMessage, this.renderForm)(recoverPasswordSuccess)}
      </Container>
    );
  }
}
