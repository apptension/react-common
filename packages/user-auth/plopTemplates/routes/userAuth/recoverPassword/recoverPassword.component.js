import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form/immutable';
import { FormattedMessage } from 'react-intl';
import { Link } from 'react-router-dom';

import { renderWhenTrueOtherwise } from '../../../utils/rendering';
import {
  SuccessMessage,
  SuccessMessageText,
} from './recoverPassword.styles';
import {
  Container,
  Header,
  Form,
  ButtonGroup,
} from '../../../components/forms/form.styles';
import messages from './recoverPassword.messages';

import { TextField } from '../../../components/forms/textField';

export class RecoverPassword extends PureComponent {
  static propTypes = {
    language: PropTypes.string,
    intl: PropTypes.object.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    recoverPassword: PropTypes.func.isRequired,
    recoverPasswordSuccess: PropTypes.bool.isRequired,
  };

  handleSubmit = (values) => new Promise(() => {
    this.props.recoverPassword(
      values.get('email'),
    );
  });

  renderForm = () => {
    const { intl, handleSubmit } = this.props;
    return (
      <Container>
        <Form onSubmit={handleSubmit(this.handleSubmit)} noValidate>
          <Field
            component={TextField}
            name="email"
            type="email"
            label={messages.emailLabel}
            intl={intl}
            messages={messages}
          />

          <ButtonGroup>
            <button type="submit">
              <FormattedMessage {...messages.submit} />
            </button>
          </ButtonGroup>
        </Form>
      </Container>
    );
  };

  renderSuccessMessage = () => {
    const { language } = this.props;
    return (
      <SuccessMessage>
        <SuccessMessageText>
          <FormattedMessage {...messages.sentMessage} />
        </SuccessMessageText>
        <Link to={`/${language}/recover-password`}>
          <FormattedMessage {...messages.resendEmail} />
        </Link>
      </SuccessMessage>
    );
  };

  render() {
    const { recoverPasswordSuccess } = this.props;
    return (
      <Container>
        <Header>
          <FormattedMessage {...messages.title} />
        </Header>
        {renderWhenTrueOtherwise(this.renderSuccessMessage, this.renderForm)(recoverPasswordSuccess)}
      </Container>
    );
  }
}
