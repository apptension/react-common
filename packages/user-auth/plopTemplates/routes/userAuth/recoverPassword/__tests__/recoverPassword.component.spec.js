import React from 'react';
import { shallow } from 'enzyme';
import { spy } from 'sinon';
import { expect } from 'chai';
import { Map } from 'immutable';
import { identity } from 'ramda';

import { RecoverPassword } from '../recoverPassword.component';
import { Form } from '../../../../components/forms/form.styles';
import { intlMock } from '../../../../utils/testUtils';

describe('<RecoverPassword />', () => {
  const defaultProps = {
    intl: intlMock(),
    handleSubmit: identity,
    recoverPassword: identity,
    recoverPasswordSuccess: false,
  };

  const component = (props) => (
    <RecoverPassword {...defaultProps} {...props} />
  );

  it('should render form', () => {
    const wrapper = shallow(component());
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render success message', () => {
    const wrapper = shallow(component({ recoverPasswordSuccess: true }));
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should call handleSubmit when form is submit', () => {
    const handleSubmit = spy();
    const wrapper = shallow(component({ handleSubmit }));
    wrapper.find(Form).simulate('submit');

    expect(handleSubmit).to.have.been.called;
  });

  it('should call recoverPassword prop when handling submit', () => {
    const email = 'email@test.com';
    const handleSubmit = spy();
    const recoverPassword = spy();

    const wrapper = shallow(component({ handleSubmit, recoverPassword }));
    wrapper.find(Form).simulate('submit');

    const [fn] = handleSubmit.getCall(0).args;

    fn(Map({ email }));

    expect(recoverPassword).to.have.been.calledWith(email);
  });
});
