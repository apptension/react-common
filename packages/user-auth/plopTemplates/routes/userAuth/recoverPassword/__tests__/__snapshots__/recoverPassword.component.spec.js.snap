// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecoverPassword /> should render form 1`] = `
<styled.div>
  <styled.h1>
    <FormattedMessage
      defaultMessage="Reset password"
      id="recoverPassword.title"
      values={Object {}}
    />
  </styled.h1>
  <styled.div>
    <styled.form
      noValidate={true}
      onSubmit={[Function]}
    >
      <Field
        component={[Function]}
        intl={
          Object {
            "formatMessage": [Function],
          }
        }
        label={
          Object {
            "defaultMessage": "Email",
            "id": "recoverPassword.fields.email.label",
          }
        }
        messages={
          Object {
            "emailFormatError": Object {
              "defaultMessage": "Email is not valid",
              "id": "recoverPassword.fields.email.error.format",
            },
            "emailLabel": Object {
              "defaultMessage": "Email",
              "id": "recoverPassword.fields.email.label",
            },
            "emailMaxLengthError": Object {
              "defaultMessage": "Email shouldn't be longer than 255 characters",
              "id": "recoverPassword.fields.email.error.maxLength",
            },
            "emailPresenceError": Object {
              "defaultMessage": "Email cannot be blank",
              "id": "recoverPassword.fields.email.error.presence",
            },
            "resendEmail": Object {
              "defaultMessage": "Resend email",
              "id": "recoverPassword.resendEmail",
            },
            "sentMessage": Object {
              "defaultMessage": "Check your email to get reset password link.",
              "id": "recoverPassword.sentMessage",
            },
            "submit": Object {
              "defaultMessage": "Send",
              "id": "recoverPassword.submit",
            },
            "title": Object {
              "defaultMessage": "Reset password",
              "id": "recoverPassword.title",
            },
          }
        }
        name="email"
        type="email"
      />
      <styled.div>
        <button
          type="submit"
        >
          <FormattedMessage
            defaultMessage="Send"
            id="recoverPassword.submit"
            values={Object {}}
          />
        </button>
      </styled.div>
    </styled.form>
  </styled.div>
</styled.div>
`;

exports[`<RecoverPassword /> should render success message 1`] = `
<styled.div>
  <styled.h1>
    <FormattedMessage
      defaultMessage="Reset password"
      id="recoverPassword.title"
      values={Object {}}
    />
  </styled.h1>
  <styled.div>
    <styled.div>
      <FormattedMessage
        defaultMessage="Check your email to get reset password link."
        id="recoverPassword.sentMessage"
        values={Object {}}
      />
    </styled.div>
    <Link
      replace={false}
      to="/undefined/recover-password"
    >
      <FormattedMessage
        defaultMessage="Resend email"
        id="recoverPassword.resendEmail"
        values={Object {}}
      />
    </Link>
  </styled.div>
</styled.div>
`;
